### PR TITLE
Rename micronaut-discovery-client module group

### PR DESCRIPTION
**Pull request by pgressa**

### CODE REVIEW

**Comment by pgressa** [comment]
Needed in order to address https://github.com/micronaut-projects/micronaut-kubernetes/issues/188

**Review by alvarosanchez — APPROVED** [review]

**Comment by pgressa** [comment]
There are some failures in tests not caused by this PR:
```
Error: eckstyle] [ERROR] /home/runner/work/micronaut-core/micronaut-core/http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java:77:5: '/*' has more than 1 empty lines before. [EmptyLineSeparator]
```
```
Error: eckstyle] [ERROR] /home/runner/work/micronaut-core/micronaut-core/http-client-core/src/main/java/io/micronaut/http/client/bind/DefaultHttpClientBinderRegistry.java:18:8: Unused import - io.micronaut.context.annotation.BootstrapContextCompatible. [UnusedImports]
```
.....



Is that intentional? @alvarosanchez @graemerocher ?

**Comment by alvarosanchez** [comment]
Sounds like someone left Checkstyle errors in the base branch

**Comment by pgressa** [comment]
> Sounds like someone left Checkstyle errors in the base branch

I'll append a commit with fixes ..

**Comment by pgressa** [comment]
Plust the test `RequestCertificateSpec` is failing on expired certificate so I'm going to update the keystore

**Comment by ilopmar** [comment]
@pgressa please take a look at what I've tried (unsuccessfully): https://github.com/micronaut-projects/micronaut-servlet/issues/75#issuecomment-703737157

**Comment by pgressa** [comment]
> Plust the test `RequestCertificateSpec` is failing on expired certificate so I'm going to update the keystore

File: http-server-netty/src/test/resources/logback.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
     </root>
 <!--    <logger name="io.micronaut.http" level="TRACE"/>-->
     <!--<logger name="io.micronaut.websocket" level="TRACE"/>-->
-<!--    <logger name="io.netty" level="TRACE"/>-->
+    <logger name="javax.net.ssl" level="TRACE"/>
+    <logger name="sun.security" level="TRACE"/>
```
Comment (graemerocher) [inline http-server-netty/src/test/resources/logback.xml]: Please remove this logback changes

**Review by graemerocher — CHANGES_REQUESTED** [review]
Does this not need to go into 2.1.x as well?

File: http-server-netty/src/test/resources/logback.xml (hunk 1 above)
Comment (pgressa) [reply http-server-netty/src/test/resources/logback.xml]: sure, forgot them ..

**Comment by pgressa** [comment]
@graemerocher any other comments?

**Pull request merged**